src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlAdvice.cs

```diff
@@ -84,7 +84,7 @@ public SamlAdvice(IEnumerable<string> references, IEnumerable<SamlAssertion> ass
         /// </summary>
         public ICollection<string> AssertionIdReferences
         {
-            get;
+            get; set;
```

Comment:
In general we don't have setters for Collections, just 'gets'.
Make this a read only property, here, below and anywhere else.
Then, squash and reissue the PR.

Thanks.